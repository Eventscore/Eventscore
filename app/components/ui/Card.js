//<Card></Card>
import React, { Component, PropTypes } from 'react';
import { Card } from 'react-native-elements';

// Consts and Libs
import { AppSizes, AppColors, AppStyles } from '@theme/';

/* Component ==================================================================== */
class CustomCard extends Component {
  static propTypes = {
    containerStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
    titleStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
  }

  static defaultProps = {
    containerStyle: [],
    titleStyle: [],
  }

  cardProps = () => {
    // Defaults
    const props = {
      dividerStyle: [{
        backgroundColor: AppColors.border,
      }],
      ...this.props,
      containerStyle: [{
        backgroundColor: AppColors.cardBackground,
        borderRadius: AppSizes.borderRadius,
        borderColor: AppColors.border,
      }],
      titleStyle: [
        AppStyles.h2,
        { marginBottom: 15 },
      ],
    };

    if (this.props.containerStyle) {
      props.containerStyle.push(this.props.containerStyle);
    }

    if (this.props.titleStyle) {
      props.titleStyle.push(this.props.titleStyle);
    }

    return props;
  }

  render = () => <Card {...this.cardProps()} />
}

/* Export Component ==================================================================== */
export default CustomCard;
