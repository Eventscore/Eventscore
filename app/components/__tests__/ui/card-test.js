/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import { Text } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Card from '@ui/Card';

it('Card renders correctly', () => {
  const tree = renderer.create(
    <Card><Text>Hello world</Text></Card>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Card w/ Title renders correctly', () => {
  const tree = renderer.create(
    <Card title={'With Title'}><Text>Hello world</Text></Card>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Card w/ Image renders correctly', () => {
  const tree = renderer.create(
    <Card image={require('../../../images/image.png')}><Text>Hello world</Text></Card>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
