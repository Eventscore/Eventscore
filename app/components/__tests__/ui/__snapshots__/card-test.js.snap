// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#D0D1D5",
          "borderRadius": 2,
        },
        Array [],
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Hello world
    </Text>
  </View>
</View>
`;

exports[`Card w/ Image renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      Object {
        "padding": 0,
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#D0D1D5",
          "borderRadius": 2,
        },
        Array [],
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={1}
        style={
          Array [
            Object {
              "flex": 1,
              "height": 150,
              "width": null,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "padding": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Hello world
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Card w/ Title renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#D0D1D5",
          "borderRadius": 2,
        },
        Array [],
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": undefined,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#43484d",
                "fontSize": 17.5,
                "fontWeight": "bold",
                "marginBottom": 15,
                "textAlign": "center",
              },
              Array [
                Object {
                  "color": "#0E4EF8",
                  "fontFamily": "HelveticaNeue",
                  "fontSize": 21,
                  "fontWeight": "800",
                  "left": 0,
                  "lineHeight": 26,
                  "margin": 0,
                  "marginBottom": 4,
                  "right": 0,
                },
                Object {
                  "marginBottom": 15,
                },
                Array [],
              ],
              undefined,
            ],
          ]
        }
      >
        With Title
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#e1e8ee",
              "height": 1,
            },
            Array [
              Object {
                "marginBottom": 15,
              },
              Array [
                Object {
                  "backgroundColor": "#D0D1D5",
                },
              ],
            ],
          ]
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Hello world
    </Text>
  </View>
</View>
`;
